Add critical new feature

dprothero opened the pull request.

dprothero commented on src/commands/zork.js:
```diff
@@ -0,0 +1,39 @@
+    if (launchZork) {
+      launchZork();
+    } else {
+      this.logger.error('I don\'t know the word "zork".');
```
If we can't get the zorkjs npm package loaded for whatever reason, we still pay a little homage using the phrasing from the game when you type a command it doesn't understand.

myfancypants replied:
the error should probably be loaded into the catch of line 27 instead of here. If a try/catch succeeds and you are only doing a truthy check on `launchZork` properly implemented there is not really much way for that lookup to fail at this point

myfancypants commented on src/commands/zork.js:
```diff
@@ -0,0 +1,44 @@
+    await super.run();
+
+    let launchZork = null;
+    try {
```
I'm not entirely sure I follow this code flow. My hunch is a lot of this code is a byproduct of not yet implementing the catch component.

If it fails to load the module, then in that catch at line 16 I would probably move lines 21-28

Because as it stands there should be no reason why line 20 `!launchZork` should ever reasonably not be true after this try/catch completes. If the subsequent catch at line 26 is hit, we should be throwing an error at that point and not letting code execution to continue

myfancypants commented on src/commands/zork.js:
```diff
@@ -0,0 +1,44 @@
+  constructor(argv, config, secureStorage) {
+    super(argv, config, secureStorage);
+    this.exec = require('await-exec');
+    this.findZork = () => require('zorkjs'); // eslint-disable-line node/no-extraneous-require,node/no-missing-require
```
What does this comment do? Is this somehow issuing commands to the linter?

dprothero replied:
Yeah... we don't have a direct dependency on the `zorkjs` module, but eslint tries to make sure you don't reference packages that aren't in your package.json. This is how you disable specific linter rules for a single line of code.

dprothero commented on src/commands/zork.js:
```diff
@@ -0,0 +1,44 @@
+  }
+}
+
+Zork.description = 'What could this be?';
```
So, these need to be properties of the _class_ as opposed to an instance object of the class. It's how oclif (the CLI framework) is designed to work.

dprothero commented:
Thanks for the quality feedback @myfancypants. I believe I've addressed the issues. Definitely agree the flow control is much easier to follow now. Take a look and let me know.
